### PR TITLE
Fix(Auth): Add custom auth android documentation

### PR DESCRIPTION
**Pull request by gpanshu**
_Issue #, if available:_

_Description of changes:_
Currently the Custom Auth flow is missing from Android documentation. This adds it.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/fragments/lib/auth/android/signin_with_custom_flow/10_cli_setup.mdx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  }
+}
+
+exports.handler = async (event) => {
+  createAuthChallenge(event);
+};
+```
+
+<Callout>
+  Note that the `sendChallengeCode` method is empty, you can use AWS service
```
Comment (sktimalsina) [inline src/fragments/lib/auth/android/signin_with_custom_flow/10_cli_setup.mdx:87]:
```suggestion
  Note that the `sendChallengeCode` method is empty. You can use AWS service
```

File: src/fragments/lib/auth/android/signin_with_custom_flow/10_cli_setup.mdx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  like SES to setup email delivery and populate the function `sendChallengeCode`
+  to send the challenge code to the user.
+</Callout>
+
+Amazon Cognito invokes the Create Auth Challenge trigger after Define Auth Challenge to create a custom challenge. In this lambda trigger you define the challenge to present to the user. `privateChallengeParameters` contains all the information to validate the response from the user. Save and close the file. Now open the file under "<your_xcode_project>/amplify/backend/function/<project_code>CreateAuthChallenge/src/package.json" and add the following:
```
Comment (sktimalsina) [inline src/fragments/lib/auth/android/signin_with_custom_flow/10_cli_setup.mdx:92]: is `xcode` typo here?

File: src/fragments/lib/auth/android/signin_with_custom_flow/30_confirmSignup.mdx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+<BlockSwitcher>
+
+<Block name="Async/Await">
+
+```swift
```
Comment (sktimalsina) [inline src/fragments/lib/auth/android/signin_with_custom_flow/30_confirmSignup.mdx:5]: swift

**Review by sktimalsina — CHANGES_REQUESTED** [review]

File: src/fragments/lib/auth/android/signin_with_custom_flow/20_signup.mdx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+<BlockSwitcher>
+
+<Block name="Async/Await">
+
+```swift
+func signUp(username: String, email: String) async {
```
Comment (div5yesh) [inline src/fragments/lib/auth/android/signin_with_custom_flow/20_signup.mdx:6]: swift code in Android page.

Comment (gpanshu) [reply src/fragments/lib/auth/android/signin_with_custom_flow/20_signup.mdx]: resolved

File: src/fragments/lib/auth/android/signin_with_custom_flow/30_confirmSignup.mdx (hunk 3 above)
Comment (gpanshu) [reply src/fragments/lib/auth/android/signin_with_custom_flow/30_confirmSignup.mdx]: resolved

File: src/fragments/lib/auth/android/signin_with_custom_flow/10_cli_setup.mdx (hunk 2 above)
Comment (gpanshu) [reply src/fragments/lib/auth/android/signin_with_custom_flow/10_cli_setup.mdx]: resolved

File: src/fragments/lib/auth/android/signin_with_custom_flow/10_cli_setup.mdx (hunk 1 above)
Comment (gpanshu) [reply src/fragments/lib/auth/android/signin_with_custom_flow/10_cli_setup.mdx:87]: resolved

**Review by tjleing — APPROVED** [review]

**Review by sktimalsina — APPROVED** [review]

File: src/fragments/lib/auth/android/signin_with_custom_flow/10_cli_setup.mdx [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+  }
+}
+
+exports.handler = async (event) => {
+  createAuthChallenge(event);
+};
+```
+
+<Callout>
+  Note that the `sendChallengeCode` method is empty. You can use AWS service
```
Comment (katiegoines) [inline src/fragments/lib/auth/android/signin_with_custom_flow/10_cli_setup.mdx:87]:
```suggestion
  Note that the `sendChallengeCode` method is empty. You can use an AWS service
```

**Review by katiegoines — APPROVED** [review]
Approved with one very nit suggestion.

**Pull request merged**